### PR TITLE
8 registration and authentication

### PR DESCRIPTION
**Pull request by rappet**

### CODE REVIEW

File: src/cljs/events.cljs [hunk 1]
```diff
@@ -19,8 +19,8 @@
...
 (rf/reg-event-db
   :http-no-on-failure
-  (fn [db fail]
+  (fn [db [_ fail]]
     (-> db
         (assoc :laedt false)
-        (assoc :error fail))))
-
+        (assoc :error fail))
+    (if (== (:status fail) 401) (set! (.-location js/window) "http://localhost:8081/api/oauth2/github") ())))
```
Comment (megajunky) [inline src/cljs/events.cljs:26]: Should register effect to handle the side effect of (set! ...)

**Review by megajunky — CHANGES_REQUESTED** [review]

File: src/cljs/events.cljs (hunk 1 above)
Comment (rappet) [reply src/cljs/events.cljs:26]: Is this really needed? The side effect would be that the SPA is terminated.

Comment (rappet) [reply src/cljs/events.cljs:26]: Do you mean [`rf/reg-event-db`](https://day8.github.io/re-frame/api-re-frame.core/#reg-fx).
I can then change it this evening.

Comment (megajunky) [reply src/cljs/events.cljs:26]: No, I meant `rf/reg-fx` and use `rf/reg-event-fx` for the :http-no-on-failure Event, with the newly defined effect (it's a bit confusing, but effects are name fx in re-frame). https://day8.github.io/re-frame/Effects/#extensible-side-effects

Comment (megajunky) [reply src/cljs/events.cljs:26]: > Is this really needed? The side effect would be that the SPA is terminated.

It's not strictly necessary, but technically we don't want our events to have side effects.

**Review by megajunky — APPROVED** [review]
It's ok to leave the side effect in the event for now

**Comment by rappet** [comment]
> It's ok to leave the side effect in the event for now

The event now returns an effect and the redirect is done in the effect handler.

**Pull request merged**